### PR TITLE
[web] remove loading indicator in -d web-server builds

### LINKED ISSUE #135226: Running with web-server doesn't remove animating flutter-loader
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Launch any flutter web app using `flutter run -d web-server`
2. Proceed to provided url
3. Notice that `flutter-loader` element (which is the animating blue line at the top) remains forever in DOM

### Expected results

Animating loader element should be removed instead of hidden by the app rendering on top.

### Actual results

Animating loader element continues to animate indefinitely causing layout shifts and making it more difficult to test performance.

### Code sample

<details><summary>Code sample</summary>

Should be reproducible on default counter app.

</details>

### Screenshots or Video

_No response_

### Logs

_No response_

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.10.6, on Microsoft Windows [Version 10.0.19044.3324], locale en-US)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.3.5)
[√] Android Studio (version 2021.3)
[√] VS Code (version 1.82.2)
[√] Connected device (3 available)
[√] Network resources

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by yjbanov**
Fixes https://github.com/flutter/flutter/issues/135226

### CODE REVIEW

File: packages/flutter_tools/lib/src/web/bootstrap.dart [hunk 1]
```diff
@@ -13,13 +13,74 @@ import 'package:package_config/package_config.dart';
...
+/// web page that's visible while the Flutter app is loading.
 String generateBootstrapScript({
   required String requireUrl,
   required String mapperUrl,
+  required bool generateLoadingIndicator,
 }) {
   return '''
 "use strict";
 
+${generateLoadingIndicator ? _generateLoadingIndicator() : ''}
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/web/bootstrap.dart:27]: does it matter that the loading indicator code will execute first now?

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/web/bootstrap.dart:27]: otherwise lgtm

Comment (yjbanov) [reply packages/flutter_tools/lib/src/web/bootstrap.dart:27]: This PR is not changing it. It was executing first before this PR.

**Review by ditman — APPROVED** [review]
LGTM! Thanks for making this behavior configurable!

File: packages/flutter_tools/lib/src/web/bootstrap.dart (hunk 1 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/web/bootstrap.dart:27]: you're right. i suck at reading diffs, LGTM

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**